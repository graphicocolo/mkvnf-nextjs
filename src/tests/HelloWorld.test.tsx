import { render, screen } from '@testing-library/react';

import '@testing-library/jest-dom';
import HelloWorld from '@/components/HelloWorld';

// 基本的なレンダリングテスト
// test('renders Hello, World!', () => {
//   render(<HelloWorld />); // コンポーネントを仮想 DOM にレンダリング
//   // screen.getByText 特定のテキストを持つ要素を検索
//   // toBeInTheDocument 要素がDOMに存在するか確認
//   expect(screen.getByText('Hello, World!')).toBeInTheDocument();
// });

// テストのグループ化
describe('Hello World レンダリングテスト', () => {
  // 基本的なレンダリングテスト
  test('要素がレンダリングされているか', () => {
    render(<HelloWorld />); // コンポーネントを仮想 DOM にレンダリング
    // screen.getByText 特定のテキストを持つ要素を検索
    // toBeInTheDocument 要素がDOMに存在するか確認
    expect(screen.getByText('Hello World')).toBeInTheDocument();
  });
});

describe('Hello World スナップショットテスト', () => {
  // コンポーネントの出力が期待通りかどうかを検証
  test('正しく出力されているか', () => {
    // レンダリングされたコンポーネントを DOM のフラグメントとして取得
    const { asFragment } = render(<HelloWorld />);
    // スナップショットテスト
    expect(asFragment()).toMatchSnapshot();
  });
});
