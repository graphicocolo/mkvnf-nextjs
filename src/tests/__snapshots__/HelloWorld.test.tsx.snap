// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hello World スナップショットテスト 正しく出力されているか 1`] = `
<DocumentFragment>
  <h1>
    Hello World
  </h1>
</DocumentFragment>
`;
